### PR TITLE
reindex monitor data

### PR DESCRIPTION
**Pull request by kjaskiewiczz**

### CODE REVIEW

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

File: client/devicemonitor/client.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mendersoftware/go-lib-micro/identity"
+	"github.com/mendersoftware/go-lib-micro/rest_utils"
+	"github.com/pkg/errors"
```
Comment (merlin-northern) [inline client/devicemonitor/client.go:26]: nit: can we sort the imports?

File: client/devicemonitor/client.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+)
+
+const (
+	defaultTimeout = time.Duration(5) * time.Second
+)
+
+// Client is the workflows client
+//go:generate ../../utils/mockgen.sh
+type Client interface {
+	CheckHealth(ctx context.Context) error
```
Comment (merlin-northern) [inline client/devicemonitor/client.go:41]: where are we calling this function?

File: client/devicemonitor/client.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+const (
+	AlertsURI = "/api/v1/devicemonitor/tenants/:tenant_id/devices/:device_id/alerts/latest"
+	HealthURI = "/api/v1/health"
+)
+
+const (
+	defaultTimeout = time.Duration(5) * time.Second
+)
+
+// Client is the workflows client
```
Comment (merlin-northern) [inline client/devicemonitor/client.go:38]: is this comment copy-paste ro do we mean it?

File: client/devicemonitor/client_test.go [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/mendersoftware/go-lib-micro/rest_utils"
```
Comment (merlin-northern) [inline client/devicemonitor/client_test.go:28]: nit: can we sort the imports?

File: api/http/api_inventory.go [hunk 5]
```diff
@@ -984,6 +986,66 @@ func (i *inventoryHandlers) GetDeviceGroupsInternalHandler(w rest.ResponseWriter
...
+
+	deviceId := r.PathParam("device_id")
+	if len(deviceId) < 1 {
+		u.RestErrWithLog(w, r, l, errors.New("device id cannot be empty"), http.StatusBadRequest)
+		return
+	}
+
+	serviceName, err := utils.ParseQueryParmStr(r, "service", false, nil)
+	// inventory service accepts only reindex requests from devicemonitor
+	if err != nil || serviceName != "devicemonitor" {
```
Comment (merlin-northern) [inline api/http/api_inventory.go:1004]: nit: could we have an array `supportedServices`?

File: model/device.go [hunk 6]
```diff
@@ -33,12 +33,15 @@ const (
...
-	AttrNameCreated  = "created_ts"
-	AttrNameTagsEtag = "tags_etag"
+	AttrScopeMonitor   = "monitor"
+
+	AttrNameID             = "id"
+	AttrNameGroup          = "group"
+	AttrNameUpdated        = "updated_ts"
+	AttrNameCreated        = "created_ts"
+	AttrNameTagsEtag       = "tags_etag"
+	AttrNameNumberOfAlerts = "number_of_alerts"
```
Comment (merlin-northern) [inline model/device.go:43]: nit, feel free to ignore, why does `alert_count` sound better to me?

File: docs/internal_api.yml [hunk 7]
```diff
@@ -241,6 +241,45 @@ paths:
...
+      tags:
+        - Internal API
+      summary: Start reindexing device attributes.
+      operationId: Start Re-indexing
+      parameters:
+        - in: query
+          name: service
+          schema:
+            type: string
+            example: "inventory"
```
Comment (merlin-northern) [inline docs/internal_api.yml:255]: technically we accept only "Devicemonitor" atm, so maybe let's not mislead in the example?

File: docs/internal_api.yml [hunk 8]
```diff
@@ -241,6 +241,45 @@ paths:
...
+        - in: path
+          name: tenant_id
+          required: true
+          description: ID of tenant owning the device.
+          schema:
+            type: string
+            example: "123456789012345678901234"
+      responses:
+        200:
+          description: Attributes has been re-indexed.
```
Comment (merlin-northern) [inline docs/internal_api.yml:273]: nit: "Attributes has" -> "Attributes _have_"

File: client/devicemonitor/client.go (hunk 1 above)
Comment (kjaskiewiczz) [reply client/devicemonitor/client.go:24]: done

File: client/devicemonitor/client.go (hunk 2 above)
Comment (kjaskiewiczz) [reply client/devicemonitor/client.go]: nowhere I guess; removed

File: client/devicemonitor/client.go (hunk 3 above)
Comment (kjaskiewiczz) [reply client/devicemonitor/client.go]: it is; fixed

File: client/devicemonitor/client_test.go (hunk 4 above)
Comment (kjaskiewiczz) [reply client/devicemonitor/client_test.go]: done

File: api/http/api_inventory.go (hunk 5 above)
Comment (kjaskiewiczz) [reply api/http/api_inventory.go:1004]: I don't feel a need for it; this is only a workaround for not having the reporting in the release so I think we'll have very nice implementation in the reporting service and having more rough one here should be OK

File: model/device.go (hunk 6 above)
Comment (kjaskiewiczz) [reply model/device.go]: as you wish ;)

File: docs/internal_api.yml (hunk 7 above)
Comment (kjaskiewiczz) [reply docs/internal_api.yml]: fixed

File: docs/internal_api.yml (hunk 8 above)
Comment (kjaskiewiczz) [reply docs/internal_api.yml]: fixed

**Review by merlin-northern — APPROVED** [review]
lgtm thanks for addressing so many suggestions.

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
I rebased it on top of master, which should fix the integration test failures

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Pull request merged**